tee-worker/litentry/pallets/identity-management/src/lib.rs

```diff
@@ -195,6 +195,28 @@ pub mod pallet {
 			Self::deposit_event(Event::IdentityRemoved { who, identity });
 			Ok(())
 		}
+
+		#[pallet::call_index(3)]
+		#[pallet::weight(15_000_000)]
+		pub fn set_identity_networks(
+			origin: OriginFor<T>,
+			who: T::AccountId,
+			identity: Identity,
+			web3networks: Vec<Web3Network>,
+		) -> DispatchResult {
+			T::ManageOrigin::ensure_origin(origin)?;
+			ensure!(IDGraphs::<T>::contains_key(&who, &identity), Error::<T>::IdentityNotExist);
+			IDGraphs::<T>::try_mutate(&who, &identity, |context| {
+				let mut c = context.take().ok_or(Error::<T>::IdentityNotExist)?;
+				ensure!(
+					identity.matches_web3networks(web3networks.as_ref()),
+					Error::<T>::WrongWeb3NetworkTypes
+				);
+				c.web3networks = web3networks;
```

Comment:
It could be worth having setter on `IdentityContext` that would also `dedup` the vector. What do you think ?

Reply:
Oh yea I'll add it, thanks

Reply:
Done :)